snapcraft.yaml

```diff
@@ -7,6 +7,77 @@ type: gadget
 architectures:
   - armhf
 confinement: strict
+grade: stable
+parts:
+  uboot:
+    plugin: make
+    source: git://git.denx.de/u-boot.git
+    source-branch: v2017.05
+    prepare: |
+      git apply ../../../uboot.patch
+      make rpi_3_32b_defconfig
+    build: |
+      if [ "$(arch)" = "x86_64" ]; then
+        if [ ! -x /usr/bin/arm-linux-gnueabihf-gcc ]; then
+          echo "ERROR: You are cross building this snap, please install gcc-arm-linux-gnueabihf"
+          exit 1
+        fi
+        CROSS_COMPILE=arm-linux-gnueabihf- make
+      else
+        make
+      fi
+    install: |
+      mkdir -p $SNAPCRAFT_PART_INSTALL/boot-assets
+      cp u-boot.bin $SNAPCRAFT_PART_INSTALL/boot-assets/uboot.bin
+      tools/mkenvimage -r -s 131072 -o $SNAPCRAFT_PART_INSTALL/boot-assets/uboot.env ../../../uboot.env.in
+      cd $SNAPCRAFT_PART_INSTALL/boot-assets; ln -s uboot.env uboot.conf
```

Comment:
This could be simplified to once call:
ln -s uboot.env $SNAPCRAFT_PART_INSTALL/boot-assets/uboot.conf